monai/transforms/transform.py

```diff
@@ -71,14 +73,29 @@ def _log_stats(data, prefix: Optional[str] = "Data"):
         raise RuntimeError(f"applying transform {transform}") from e
 
 
-class Randomizable(ABC):
+class ThreadUnsafe:
+    """
+    A class to denote that the transform will mutate its member variables,
+    when being applied. Transforms inheriting this class should be used
+    cautiously in a multi-thread context.
+
+    This type is typically used by :py:class:`monai.data.CacheDataset` and
+    its extensions, where the transform cache is built with multiple threads.
+    """
+
+    pass
+
+
+class Randomizable(ABC, ThreadUnsafe):
```

Comment:
May I know why Randomizable transforms are thread-unsafe?
I think "writing class member variables at runtime" causes the `thread-unsafe` issue, right?

Thanks.

Reply:
I think calling `self.R` would implicitly change its state here, so it's not thread-safe.

Reply:
Could you please help enhance the doc-string to describe this `self.R` point?

Thanks.